hours/tests/test_resource_periods_as_text.py

```diff
@@ -241,6 +241,48 @@ def test_rule_as_text_frequency_ordinal(context, subject, start, ordinal, modifi
     assert rule_as_text_en != rule_as_text_fi
 
 
+@pytest.mark.parametrize(
+    "start,end,expect_start,expect_end",
+    [
+        (datetime.date(2023, 1, 1), None, False, False),
+        (datetime.date(2023, 1, 1), datetime.date(2023, 3, 1), True, True),
+        (None, None, False, False),
+        (None, datetime.date(2023, 3, 1), False, True),
+        (datetime.date(2023, 3, 1), None, True, False),
+        (datetime.date(2023, 3, 1), datetime.date(2023, 3, 31), True, True),
+    ],
+)
+@pytest.mark.django_db
+def test_date_period_as_text_content(resource, start, end, expect_start, expect_end):
+    after_start_date = datetime.date(2023, 2, 1)
+    date_period = DatePeriodFactory(
+        name="Test hours",
+        resource=resource,
+        resource_state=State.OPEN,
+        start_date=start,
+        end_date=end,
+    )
+
+    with translation.override("fi"):
+        text = date_period.as_text(after_start_date)
+
+        start_str = formats.date_format(start) if start else ""
+        end_str = formats.date_format(end) if end else ""
```

Comment:
Using `formats.date_format` makes assumptions about the implementation details of the unit you're testing and can potentially hide some bugs. For example:
- if `formats` introduced a bug in some update that would specifically break the `date_format` function (e.g. returns "foo" instead of a proper date), this test wouldn't catch it
- if `formats` was completely removed from the project, this test would break